gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -218,38 +219,64 @@ def run(argv=None):
-  if variant_merger:
-    if known_args.optimize_for_large_inputs:
-      partitions = variants | 'PartitionVariants' >> beam.Partition(
-          partition_variants.PartitionVariants(partitioner),
-          partitioner.get_num_partitions())
-      merged = []
-      for i in xrange(partitioner.get_num_partitions()):
-        merged.append(partitions[i] | 'MergeVariants' + str(i) >>
+  if partitioner:
+    num_partitions = partitioner.get_num_partitions()
+    partitioned_variants = variants | 'PartitionVariants' >> beam.Partition(
+        partition_variants.PartitionVariants(partitioner), num_partitions)
+    variants = []
+    for i in range(num_partitions):
+      if partitioner.should_keep_partition(i):
+        variants.append(partitioned_variants[i])
+      else:
+        #  An extra dummy partition has been added for residual variants. This
+        #  extra partition will be ignored in the rest of the pipeline.
+        num_partitions -= 1
+  else:
+    # By default we don't partition data, so we have only 1 partition.
+    num_partitions = 1
+    variants = [variants]
+
+  for i in range(num_partitions):
+    suffix = ''
+    if partitioner and partitioner.get_partition_name(i):
+      suffix = '_' + partitioner.get_partition_name(i)
+    if variant_merger:
+      variants[i] |= ('MergeVariants' + suffix >>
                       merge_variants.MergeVariants(variant_merger))
-      variants = merged | 'FlattenPartitions' >> beam.Flatten()
-    else:
-      variants |= ('MergeVariants' >> merge_variants.MergeVariants(
-          variant_merger))
-  proc_variants = variants | 'ProcessVaraints' >> beam.Map(
-      processed_variant_factory.create_processed_variant).\
-    with_output_types(processed_variant.ProcessedVariant)
-  _ = (proc_variants |
-       'VariantToBigQuery' >> variant_to_bigquery.VariantToBigQuery(
-           known_args.output_table,
-           header_fields,
-           variant_merger,
-           processed_variant_factory,
-           append=known_args.append,
-           allow_incompatible_records=known_args.allow_incompatible_records,
-           omit_empty_sample_calls=known_args.omit_empty_sample_calls,
-           num_bigquery_write_shards=known_args.num_bigquery_write_shards))
+    variants[i] |= (
+        'ProcessVaraints' + suffix >>
+        beam.Map(processed_variant_factory.create_processed_variant).\
+            with_output_types(processed_variant.ProcessedVariant))
+  # If this is auto partitioning (i.e. no config is given) flatten partitions.
+  if partitioner and partitioner.should_flatten():
+    variants = [variants | 'FlattenPartitions' >> beam.Flatten()]
+    num_partitions = 1
+
+  for i in range(num_partitions):
+    suffix = ''
+    if partitioner and partitioner.get_partition_name(i):
+      suffix = '_' + partitioner.get_partition_name(i)
+    _ = (variants[i] | 'VariantToBigQuery' + suffix >>
+         variant_to_bigquery.VariantToBigQuery(
+             known_args.output_table + suffix,
```

Comment:
curious: are there any restrictions on the naming here? Just wondering if someone has a very long or strange suffix (ok for table name) that it wont break Dataflow. Again, probably just ok to use `i` here as well, but up to you.
nit: if you do decide to use `i`, then I recommend changing above to be `table_name = known_args.output_table` and then do `table_name += ...` in the if statement.